src/fsw/FCCode/Drivers/Piksi.cpp

```diff
@@ -271,32 +309,45 @@ unsigned char Piksi::read_all() {
     #ifdef DESKTOP
     return _read_return;
     #else
+
+    noInterrupts();
+
+    int bytes = Serial4.available();
+    unsigned long time = interrupt_count;
+
+    // from check_bytes, but set last_bytes to 0
+    if (bytes > last_bytes) {
+
+        if (time - last_time > 100){
+            start_time = time;
+        }
+
+      last_time = time;
+    }
+    last_bytes = 0;
+
+    buffer_begin = buffer;
+    buffer_end = buffer;
+    // copy buffer up to available
+    for (int i = 0; i < bytes; i++){
+        *(buffer_end++) = Serial4.read();
+    }
+    clear_bytes();
+
+    microdelta = (time - start_time)*100; //convert to us
+
+    interrupts();

```

Comment:
Also, now because the timing is strictly tied to the interrupt schedule I _think_ we should actually get rid of the timing code here. Like it doesn't allow us to gain a better estimate of the time the messages arrived so we may as well just leave it be other than the `last_bytes` value if that makes sense:
```suggestion
    noInterrupts();

    int bytes = Serial4.available();
    last_bytes = 0;

    buffer_begin = buffer;
    buffer_end = buffer;
    // copy buffer up to available
    for (int i = 0; i < bytes; i++){
        *(buffer_end++) = Serial4.read();
    }
    clear_bytes();

    microdelta = interrupt_count*100; //convert to us

    interrupts();

```
Also, with the small booking keeping change in the ISR, I think we can define the ` microdelta` just based on the interrupt count.